pandas/core/arrays/numpy_.py

```diff
@@ -0,0 +1,327 @@
+import numpy as np
+
+from pandas._libs import lib
+
+from pandas.core.dtypes.common import extract_array
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+from pandas.core.dtypes.inference import is_list_like
+
+from pandas import compat
+from pandas.core import nanops
+
+from .base import ExtensionArray, ExtensionOpsMixin
+
+
+class PandasDtype(ExtensionDtype):
+    """
+    A Pandas ExtensionDtype for NumPy dtypes.
+
+    .. versionadded:: 0.24.0
+
+    This is mostly for internal compatibility, and is not especially
+    useful on its own.
+
+    Parameters
+    ----------
+    dtype : numpy.dtype
+    """
+    _metadata = ('_dtype',)
+
+    def __init__(self, dtype):
+        dtype = np.dtype(dtype)
+        self._dtype = dtype
+        self._name = dtype.name
+        self._type = dtype.type
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def _is_numeric(self):
+        # exclude object, str, unicode, void.
+        return self.kind in set('biufcmM')
+
+    @property
+    def _is_boolean(self):
+        return self.kind == 'b'
+
+    @classmethod
+    def construct_from_string(cls, string):
+        return cls(np.dtype(string))
+
+    def construct_array_type(cls):
+        return PandasArray
+
+    @property
+    def kind(self):
+        return self._dtype.kind
+
+    @property
+    def itemsize(self):
+        return self._dtype.itemsize
+
+
+class PandasArray(ExtensionArray, ExtensionOpsMixin):
+    """
+    A pandas ExtensionArray for NumPy data.
+
+    .. versionadded :: 0.24.0
+
+    This is mostly for internal compatibility, and is not especially
+    useful on its own.
+
+    Parameters
+    ----------
+    values : ndarray
+        The NumPy ndarray to wrap. Must be 1-dimensional.
+    """
+    # If you're wondering why pd.Series(cls) doesn't put the array in an
+    # ExtensionBlock, search fo `ABCPandasArray`. We check for
+    # that _typ to ensure that that users don't unnecessarily use EAs.
+    _typ = "npy_extension"
+    __array_priority__ = 1000
+
+    # ------------------------------------------------------------------------
+    # Constructors
+
+    def __init__(self, values):
+        if isinstance(values, type(self)):
+            values = values._ndarray
+        values = np.asarray(values)
+        if values.ndim != 1:
+            raise ValueError("PandasArray must be 1-dimensional.")
+
+        self._ndarray = values
+        self._dtype = PandasDtype(values.dtype)
+
+    @classmethod
+    def _from_sequence(cls, scalars, dtype=None, copy=False):
+        # we deliberately ignore dtype to not deal with casting issues.
+        # That's probably a bad idea.
```

Comment:
Do you mean the ignoring is bad, or otherwise casting would be bad ;)

Reply:
Ignoring it :) Looking more closely... the failures are all around int vs. float. If we use a float for the test array, then things pass.